sdk/translation/Azure.AI.Translation.Text/src/Custom/TextTranslationTranslateOptions.cs

```diff
@@ -0,0 +1,175 @@
+        /// from a customized system built with Custom Translator. Add the Category ID from your Custom Translator
+        /// project details to this parameter to use your deployed customized system. Default value is: general.
+        /// </param>
+        /// <param name="profanityAction">
+        /// Specifies how profanities should be treated in translations.
+        /// Possible values are: NoAction (default), Marked or Deleted.
+        /// </param>
+        /// <param name="profanityMarker">
+        /// Specifies how profanities should be marked in translations.
+        /// Possible values are: Asterisk (default) or Tag.
+        /// </param>
+        /// <param name="includeAlignment">
+        /// Specifies whether to include alignment projection from source text to translated text.
+        /// Possible values are: true or false (default).
+        /// </param>
+        /// <param name="includeSentenceLength">
+        /// Specifies whether to include sentence boundaries for the input text and the translated text.
+        /// Possible values are: true or false (default).
+        /// </param>
+        /// <param name="suggestedFrom">
+        /// Specifies a fallback language if the language of the input text can&apos;t be identified.
+        /// Language autodetection is applied when the from parameter is omitted. If detection fails,
+        /// the suggestedFrom language will be assumed.
+        /// </param>
+        /// <param name="fromScript"> Specifies the script of the input text. </param>
+        /// <param name="toScript"> Specifies the script of the translated text. </param>
+        /// <param name="allowFallback">
+        /// Specifies that the service is allowed to fall back to a general system when a custom system doesn&apos;t exist.
+        /// Possible values are: true (default) or false.
+        ///
+        /// allowFallback=false specifies that the translation should only use systems trained for the category specified
+        /// by the request. If a translation for language X to language Y requires chaining through a pivot language E,
+        /// then all the systems in the chain (X → E and E → Y) will need to be custom and have the same category.
+        /// If no system is found with the specific category, the request will return a 400 status code. allowFallback=true
+        /// specifies that the service is allowed to fall back to a general system when a custom system doesn't exist.
+        /// </param>
+        public TextTranslationTranslateOptions(IEnumerable<string> targetLanguages, IEnumerable<string> content, string clientTraceId = null, string sourceLanguage = null, TextType? textType = null, string category = null, ProfanityAction? profanityAction = null, ProfanityMarker? profanityMarker = null, bool? includeAlignment = null, bool? includeSentenceLength = null, string suggestedFrom = null, string fromScript = null, string toScript = null, bool? allowFallback = null): base()
+        {
```

Comment:
Have you checked whether `Options` are usually used for validation of parameters that are passed in? For example - shall we check for target languages to be "valid" (non empty?)?

Reply:
Yes.  The pattern I see in the other services is to use the Options class only as a property bag.

Reply:
I have added argument assertions in the constructors for required parameters.